src/dune/dune_file.ml

```diff
@@ -2191,23 +2211,29 @@ module Deprecated_library_name = struct
   type t =
     { loc : Loc.t
     ; project : Dune_project.t
-    ; old_public_name : Public_lib.t
-    ; new_public_name : Lib_name.t
+    ; old_public_name : bool * Public_lib.t
```

Comment:
It's not obvious what this `bool` mean without reading the parsing code. We should use a dedicated type or a record.

Reply:
Fixed, I defined a new record type.